AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/Storage.smithy

```diff
@@ -0,0 +1,275 @@
+// SPDX-License-Identifier: Apache-2.0
+namespace aws.cryptography.keyStore
+
+
+//= aws-encryption-sdk-specification/framework/key-store/encrypted-key-store.md#type
+//= type=implication
+//# A union that MUST hold the following three options
+//#
+//#- ActiveHierarchicalSymmetricVersion [HierarchicalSymmetricVersion](#hierarchicalsymmetricversion)
+//#- HierarchicalSymmetricVersion [HierarchicalSymmetricVersion](#hierarchicalsymmetricversion)
+//#- ActiveHierarchicalSymmetricBeacon
+@documentation("Configures the kind of key that an encrypted blob represents.")
+union BranchKeyType {
+  @documentation("The version the active branch key version. This version is used to encrypt messages")
+  ActiveHierarchicalSymmetricVersion: String,
+  //= aws-encryption-sdk-specification/framework/key-store/encrypted-key-store.md#hierarchicalsymmetricversion
+  //= type=implication
+  //# A structure that MUST have one member,
+  //# the UTF8 Encoded value of the version of the branch key.
+  @documentation("The version for a decrypt only branch key version. These versions are used to decrypt messages.")
+  HierarchicalSymmetricVersion: String,
+  @documentation("The information regarding a symmetric beacon key.")
+  ActiveHierarchicalSymmetricBeacon: ActiveHierarchicalSymmetricBeacon
+}
+
+@documentation("Information regarding a symmetric beacon key. At this time there is no additional information.")
+structure ActiveHierarchicalSymmetricBeacon {}
+
+
+//= aws-encryption-sdk-specification/framework/key-store/encrypted-key-store.md#encryptedhierarchicalkey
+//= type=implication
+//# This structure MUST include all of the following fields:
+//#
+//#- [BranchKeyId](./structures.md#branch-key-id)
+//#- [Type](#type)
+//#- CreateTime: Timestamp in ISO 8601 format in UTC, to microsecond precision.
+//#- KmsArn: The AWS KMS Key ARN used to protect the CiphertextBlob value.
+//#- [EncryptionContext](./structures.md#encryption-context-3)
+//#- CiphertextBlob: The encrypted binary for the hierarchical key.
+@documentation("Information about an encrypted hierarchical key. This abstracts the structure of this information from the underlying storage.")
+structure EncryptedHierarchicalKey {
+  @required
+  @documentation("The identifier for this encrypted key.")
+  Identifier: String,
+
+  @required
+  @documentation("The type of encrypted key.")
+  Type: BranchKeyType,
+
+  @required
+  @documentation("The create time as an ISO 8061 UTC string.")
+  CreateTime: String,
+
+  @required
+  @documentation("The KMS ARN used to encrypt this encrypted key.")
+  KmsArn: String,
+
+  @required
+  @documentation("The encryption context needed to decrypt this encrypted key.")
+  EncryptionContext: EncryptionContextString,
+
+  @required
+  @documentation("The ciphertext for this encrypted key.")
+  CiphertextBlob: Blob,
+}
+
+map EncryptionContextString {
+  key: String,
+  value: String,
+}
+
+@aws.polymorph#extendable
+resource EncryptedKeyStore {
```

Comment:
Bikeshedding on name for the Key Store’s Storage interface, 
which is currently called EncryptedKeyStore, 
which Tony is against because… we use Key Store far too often.
- Storage
- Backing Table
- Table

I think I really like Table...

Reply:
@seebees and I synced on a call, 
EncryptedTable.